### PR TITLE
added a support for a single GPU selection

### PR DESCRIPTION
**Pull request by evstigneevnm**
This adds an option to select and burn a single GPU if many GPUs are installed on a server/desktop and other GPUs are used by other users and they must not be burned.

### CODE REVIEW

**Comment by evstigneevnm** [comment]
This is a needed option for my server because we had 4 teslas and now 2 more installed. I needed to check only these two but not to check other four because i ran other jobs on the other 4.

File: gpu_burn-drv.cpp [hunk 1]
```diff
@@ -695,6 +741,28 @@ int main(int argc, char **argv) {
...
+		{
+			int count = 0;
+			checkError(cuInit(0));
+			checkError(cuDeviceGetCount(&count));
+			if(count == 0)
+    		{
+        		throw std::runtime_error("There is no compartable device found!\n");
+    		}
+    		for (int i_dev = 0; i_dev < count; i_dev++) 
+            {
```
Comment (wilicc) [inline gpu_burn-drv.cpp:755]: Indentation

File: gpu_burn-drv.cpp [hunk 2]
```diff
@@ -695,6 +741,28 @@ int main(int argc, char **argv) {
...
+				CUdevice device_l;
+				char device_name[255];
+				checkError(cuDeviceGet ( &device_l, i_dev ) );
+				checkError(cuDeviceGetName ( device_name, 255, device_l) ); 
+				size_t device_mem_l;
+				checkError(cuDeviceTotalMem ( &device_mem_l, device_l ) );
+				printf("ID:%i, %s, mem:%dMB \n", i_dev, device_name, device_mem_l/1000/1000 );//,device_prop.pciBusID,device_prop.pciDeviceID,device_prop.pciDomainID);
```
Comment (wilicc) [inline gpu_burn-drv.cpp:762]: Commented out code to be removed

File: gpu_burn-drv.cpp [hunk 3]
```diff
@@ -695,6 +741,28 @@ int main(int argc, char **argv) {
...
+			checkError(cuDeviceGetCount(&count));
+			if(count == 0)
+    		{
+        		throw std::runtime_error("There is no compartable device found!\n");
+    		}
+    		for (int i_dev = 0; i_dev < count; i_dev++) 
+            {
+				CUdevice device_l;
+				char device_name[255];
+				checkError(cuDeviceGet ( &device_l, i_dev ) );
```
Comment (wilicc) [inline gpu_burn-drv.cpp:758]: Inconsistent spaces near brackets, see how they are used elsewhere in the code

File: gpu_burn-drv.cpp [hunk 4]
```diff
@@ -729,6 +796,25 @@ int main(int argc, char **argv) {
                 exit(1);
             }
 		}
+		if (argc >= 2 && strncmp(argv[i],"-i",2)==0)
+		{
+            thisParam++;
```
Comment (wilicc) [inline gpu_burn-drv.cpp:801]: Indentation

File: gpu_burn-drv.cpp [hunk 5]
```diff
@@ -729,6 +796,25 @@ int main(int argc, char **argv) {
...
+
+            // -mNNN[%]
+            // -m NNN[%]
+            if(argv[i][2]){
```
Comment (wilicc) [inline gpu_burn-drv.cpp:805]: Spacing near brackets inconsistent

File: gpu_burn-drv.cpp [hunk 6]
```diff
@@ -729,6 +796,25 @@ int main(int argc, char **argv) {
...
+                device_id = atoi(argv[i]+2);
+                printf("using only device id = %i\n", device_id);
+            } else if(i+1 < args.size()){
+                i++;
+			    thisParam++;
+                device_id = atoi(argv[i]);
+                printf("using only device id = %i\n", device_id);
+            }else{
```
Comment (wilicc) [inline gpu_burn-drv.cpp:813]: Spacing near brackets inconsistent

**Review by wilicc — CHANGES_REQUESTED** [review]

**Comment by wilicc** [comment]
> This adds an option to select and burn a single GPU if many GPUs are installed on a server/desktop and other GPUs are used by other users and they must not be burned.

This is a nice change.  Could you fix some typos/code formatting/indentation?
I'll then give this a spin and if it seems to work, I'll merge.

**Comment by evstigneevnm** [comment]
Thank you for replying so fast.
I think i fixed it.

**Comment by evstigneevnm** [comment]
Think it's done.

**Pull request closed without merging**